src/css/jwplayer/flags/floatingplayer.less

```diff
@@ -20,16 +20,40 @@
     }
 
     .jw-float-icon {
-        display: block;
+        pointer-events: all;
+        cursor: pointer;
+        display: none;
+
+        .jw-svg-icon {
+            filter: drop-shadow(0 0 1px #000);
+        }
+    }
+
+    &.jw-floating-dismissible .jw-dismiss-icon {
+        display: none;
+    }
+}
+
+.jw-flag-floating.jw-floating-dismissible:not(.jw-flag-controls-hidden) {
+    &.jw-state-paused,
+    &:not(.jw-state-paused):not(.jw-flag-user-inactive) {
+        .jw-float-icon {
+            display: flex;
+        }
+
+        .jw-logo {
+            display: none;
+        }
```

Comment:
@robwalch Cleaned up the css a little bit, but still looks messy because of `not`. The aim is to hide the close icon (`jw-float-icon`) when control bar disappears, and hide the logo only when floating icon is showing up while floating.

Reply:
So can you make the flags exclusive? When we hide controls we remove the controls module, that should also remove the `jw-floating-dismissible` flag. So can make it impossible for the `jw-flag-controls-hidden` and `jw-floating-dismissible` co-exists. If they are exclusive of each other you never have to add this `not` rule.

Reply:
Also why do you need `:not(.jw-state-paused)`? It's paused or it's not inactive.